hawkular-alerts-actions-api/src/main/java/org/hawkular/alerts/actions/api/ActionPlugin.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.alerts.actions.api;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Define an alerts actions plugin implementation
+ * Plugin must have a valid name that will be used at registration phase
```

Comment:
Good question.
At the moment there is not any strict validation, the plugin must be just a string.
But I think that in the future we can add some regexp for valid (i.e. not starting with number, special characters, etc).
So, no restriction on this version now.